orodha-keycloak/orodha_keycloak/connection.py

```diff
@@ -0,0 +1,129 @@
+                password(str): The password of the new user.
+
+        Returns:
+            new_user: The new user info genereated by the keycloak server.
+
+        """
+        new_user = self.admin_connection.create_user(
+            {
+                "email": user_info.get("email"),
+                "username": user_info.get("username"),
+                "enabled": True,
+                "firstName": user_info.get("firstName"),
+                "lastName": user_info.get("lastName"),
+                "credentials": [
+                    {
+                        "value": user_info.get("password"),
+                        "type": "password",
+                    }
+                ],
+            },
+            exist_ok=False,
+        )
+
+        return new_user
+
+    def delete_user(self, user_id):
+        """
+        Deletes a keycloak user with a given user_id.
+
+        Args:
+            user_id(str): The user id of the user to be deleted.
+
+        Returns:
+            response: The response from the keycloak server with info about the deletion.
+        """
+        response = self.admin_connection.delete_user(user_id=user_id)
+
+        return response
+
+    def check_if_user_exists(self, user_identification, is_token=False):
+        """
+        Takes either a user_id or username and checks if a certain user exists.
+
+        Args:
+            user_identification(str|dict): Can either be a user_id or a JWT token,
+                is used to access keycloak in a query.
+            is_token(bool): A boolean value that determines if we use user_id or
+                token in our keycloak call. If True, we use the user_identification
+                variable as if it were a JWT token.
+
+        Returns:
+            user: The user, if any, that is associated with this user_identification value.
+        """
+
+        #NOTE: This function will be changed to return only relevant information from the decoded
+        #   token once I determine the "shape" of the response data.
+        if is_token:
+            return_value = self._decode_jwt(user_identification)
+        else:
+            return_value = self.admin_connection.get_user(user_identification)
+
+        return return_value
+
+    def _decode_jwt(self, token):
+        """
+        Small helper function which decodes a JWT token using the client connection.
+
+        Args:
+            token(dict): A JWT token that we get from keycloak.
+
+        Returns:
+            token_info(dict): The decoded information from the token.
+        """
+        keycloak_public_key = "-----BEGIN PUBLIC KEY-----\n" + \
+            self.client_connection.public_key() + "\n-----END PUBLIC KEY-----"
+        options = {
+            "verify_signature": True,
+            "verify_aud": True,
+            "verify_exp": True
+        }
+        token_info = self.client_connection.decode_token(
+            token['access_token'], key=keycloak_public_key, options=options)
```

Comment:
@Edison-Stuart Assume that `token` is the function argument *is* the access token; this assumes a dictionary containing the token.